app/services/xokul/connection.rb

```diff
@@ -4,38 +4,26 @@ module Xokul
   module Connection
     BASE_URL     = Tenant.configuration.api_host
     BEARER_TOKEN = Tenant.credentials.dig(:xokul, :bearer_token)
+    OPTIONS      = {
+      headers:     { 'Authorization' => "Bearer #{BEARER_TOKEN}" },
+      use_ssl:     true,
+      verify_mode: OpenSSL::SSL::VERIFY_PEER
+    }.freeze
 
-    private_constant :BASE_URL, :BEARER_TOKEN
+    private_constant :BASE_URL, :BEARER_TOKEN, :OPTIONS
 
-    # rubocop:disable Metrics/MethodLength
-    def self.request(path, params: {}, **http_options)
-      http_options[:open_timeout] ||= http_options[:read_timeout] ||= 10 if Rails.env.test?
-
-      response = Support::RestClient.get(
-        URI.join(BASE_URL, path).to_s,
-        headers:     {
-          'Authorization' => "Bearer #{BEARER_TOKEN}",
-          'Content-Type'  => 'application/json'
-        },
-        payload:     params.to_json,
-        use_ssl:     true,
-        verify_mode: OpenSSL::SSL::VERIFY_PEER,
-        **http_options
+    def self.request(path, params: {}, **options)
+      resp = Support::RestClient.get(
+        URI.join(BASE_URL, path).to_s, payload: params.to_json, **OPTIONS.merge(options)
       )
 
-      response.error!
-
-      unmarshal = response.unmarshal_json
-      case unmarshal
-      when Array
-        unmarshal.map(&:deep_symbolize_keys)
-      when Hash
-        unmarshal.deep_symbolize_keys
-      else
-        unmarshal
-      end
+      resp.error!
+      resp.decode
+    rescue Net::HTTPError, Net::HTTPFatalError => e
```

Comment:
Sıklıkla ortaya çıkabilecek network kaynaklı  hataları, örneğin timeout'ları bence Xokul'da ve/veya burada yakalayıp Rollbar için error değil bir başka şey örneğin warning üretelim.  Rollbar error'leri doğrudan bug issue açıyor, ondan kaçınmak için.